### ISSUE TITLE
Emoji Recognition using emoji.traineddata is not at all accurate while detecting emoji in chat images using Tesseract 4

### ISSUE THREAD

**Issue by Avani1994** [Shreeshrii/tessdata_emoji#3]
Hi @Shreeshrii, I am using  emoji.traineddata from emojilegacy folder to recognize emojis and text  in Whatsapp/Facebook Chat Images. So while recognizing I am using this command :
pytesseract.image_to_string (Image.open('guguemoji.jpeg'), lang=("emoji+eng")). However the performance is very poor. It could not get a single emoji right. But detects english language text very well. I think I am missing something on combining two models, or will I need to fine tune / cut off the few top layers from english model in tessdata_best myself to retrain it for emoji data? Can you please suggest a possible way of atleast increasing the accuracy for ("emoji + eng").

**Comment by Shreeshrii** [comment]
That was an experimental traineddata. I haven't really used it.

You can try fine-tuning eng  if you have some specific emojis that you want
to recognise.

**Comment by Avani1994** [comment]
Hey Thanks, @Shreeshrii, I just wanted some guidance from you, so we can create custom tif and box files using tesseract 3 and can train a few specific emoji, but with LSTMs can only use synthetic data generated using text2image program. I have following questions:

1. What do you think should we be using for our use case, should we go with basic model or should go with lstm one. What do you think would be more accurate?

2. Also if we want to go with Tessercat 4 (Lstms), can you please guide on from how should I create emoji training data, like should we take pics for emoji (whatsapp chat emoji images (one per text line)) and then create corresponding box files using lstmbox as we cant use tesstrain.sh because we don't have any language as emoji (--lang emoji) in base models. 

3. What I am more concerned is since Deep Neural nets require a lots of data to get trained and produce some good results, Will it be possible that we fine tune it with few specific emojis and it will recognize them accurately? Basically I want to know how much train data will be needed to fine tune english model to be able to recognize some specific emojis. If you could suggest some starting point it will be helpful.

**Comment by Shreeshrii** [comment]
Language code doesn't matter, you can do training as eng and at end create
traineddata with a different name.

One character per line won't give very good results with lstmtraining.

It is possible that tesseract 3 or legacy engine will give better results
for this. You have to try it out. Legacy training doesn't take too long.

Have you seen tesstrain repo. Using a makefile you can train using single
line images and their groundtruth.

**Comment by Avani1994** [comment]
Thanks @Shreeshrii will definitely try with Tesseract 3. However, for Tesseract 4 I saw tesstrain repo. Though seems like providing groundtruth for line images is a big task for emojis as tesseract currently does not detect emojis well. So other than doing it manually, are you aware of any other approach? Since you have already done emoji training, it will be helpful to know how you came up with ground truth for emoji line images.
